src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -298,7 +298,7 @@ private void UpdateSyncModes(SyncMode newModes, string? reason = null)
         private bool IsInAStickyFullSyncMode(Snapshot best)
         {
             long bestBlock = Math.Max(best.Processed, LastBlockThatEnabledFullSync ?? 0);
-            bool hasEverBeenInFullSync = bestBlock > _pivotNumber && best.State > _pivotNumber;
```

Comment:
pivots move and can move in front of bestblock etc

Reply:
I think this change makes sense. I'm wondering if we can write unit tests to cover it

Reply:
Existing tests cover it? But don't cover moving pivots?

Reply:
What I mean write a test that is failing for the previous version of code and it passes after fix

Reply:
Made issue for test improvement https://github.com/NethermindEth/nethermind/issues/5734